src/tools/plan-generator.ts

```diff
@@ -109,9 +109,13 @@ export class PlanGenerator {
     }
   }
 
-  /** Generates a Kotlin `StorageKey` from a StorageKey. */
-  createStorageKey(storageKey: StorageKey | undefined): string {
-    return `StorageKeyParser.parse("${(storageKey || '').toString()}")`;
+  /** Generates a Kotlin `StorageKey` from a recipe Handle. */
+  createStorageKey(handle: Handle): string {
+    const storageKey = handle.storageKey;
+    if (!storageKey && handle.fate === 'create') {
+      return ktUtils.applyFun('CreateableStorageKey', [quote(handle.id)]);
+    }
+    return ktUtils.applyFun('StorageKeyParser.parse', [(storageKey || '').toString()]);
```

Comment:
`storageKey || ''` - the ||'' isn't needed,  the storageKey must not be empty at this point.

nitnit - the local variable storageKey seems redundant, can just use handle.storageKey, i think